persistent_connection.go

```diff
@@ -177,6 +180,17 @@ func (pconn *persistentConn) readResponse() (int, string, error) {
 			err:       fmt.Errorf("error reading response: %s", err),
 			temporary: true,
 		}
+	} else if code == replyConnectionClosed {
+		pconn.controlConn.SetReadDeadline(time.Now().Add(pconn.config.Timeout))
+		code, msg, err = pconn.reader.ReadResponse(0)
```

Comment:
I think it is the user of the data connection's responsibility to read the extra control connection response (even in the case of data connection being closed early). I think a sign this doesn't belong here is that it is basically duplicates all of the readResponse() method itself.